### ISSUE TITLE
feat: StepLR and MultiStepLR learning rate schedulers

### ISSUE THREAD

**Issue by Samyssmile** [Samyssmile/edux#212]
## Summary

Only `CosineWarmupScheduler` and constant LR are available. Classic ImageNet training (ResNet paper) uses **StepLR** (multiply LR by 0.1 every 30 epochs). Multi-step and exponential decay schedulers are also standard options.

## Current State

```java
// LrScheduler.java — only two options
public interface LrScheduler {
    float lr(int step, int totalSteps);
    
    static LrScheduler cosineWarmup(float baseLr, float minLr, int warmupSteps) { ... }
    static LrScheduler constant(float lr) { ... }
}
```

### 1. StepLR

Multiply LR by a factor at regular intervals:

```java
/**
 * Decays learning rate by gamma every stepSize epochs.
 * Classic ResNet: stepSize=30, gamma=0.1 (÷10 every 30 epochs)
 */
public static LrScheduler stepDecay(float baseLr, int stepSizeInSteps, float gamma) {
    return (step, totalSteps) -> {
        int numDecays = step / stepSizeInSteps;
        return baseLr * (float) Math.pow(gamma, numDecays);
    };
}
```

### 2. MultiStepLR

Decay at specific milestones:

```java
/**
 * Decays learning rate by gamma at each milestone step.
 * Example: milestones=[30*391, 60*391, 80*391], gamma=0.1
 *          (decay at epoch 30, 60, 80 for 391 batches/epoch)
 */
public static LrScheduler multiStep(float baseLr, int[] milestoneSteps, float gamma) {
    return (step, totalSteps) -> {
        int numDecays = 0;
        for (int milestone : milestoneSteps) {
            if (step >= milestone) numDecays++;
        }
        return baseLr * (float) Math.pow(gamma, numDecays);
    };
}
```

### 3. ExponentialLR

Smooth exponential decay:

```java
/**
 * Exponential decay: lr = baseLr * gamma^step
 */
public static LrScheduler exponential(float baseLr, float gamma) {
    return (step, totalSteps) -> baseLr * (float) Math.pow(gamma, step);
}
```

### 4. Warmup Wrapper (composable)

```java
/**
 * Adds linear warmup to any scheduler.
 */
public static LrScheduler withWarmup(LrScheduler base, float baseLr, int warmupSteps) {
    return (step, totalSteps) -> {
        if (step < warmupSteps) {
            return baseLr * step / warmupSteps;
        }
        return base.lr(step - warmupSteps, totalSteps - warmupSteps);
    };
}
```

### Usage — Classic ResNet-90 baseline

```java
// SGD + StepLR: the original ResNet ImageNet recipe
var optimizer = new SGD(0.1f, 0.9f, 1e-4f, false);
int stepsPerEpoch = trainLoader.size();  // need to expose this
var scheduler = LrScheduler.withWarmup(
    LrScheduler.stepDecay(0.1f, 30 * stepsPerEpoch, 0.1f),
    0.1f,
    5 * stepsPerEpoch  // 5 epoch warmup
);

var trainer = new Trainer(model, loss, optimizer)
    .withScheduler(scheduler);
```

## Acceptance Criteria

- [ ] `LrScheduler.stepDecay(baseLr, stepSize, gamma)` implemented
- [ ] `LrScheduler.multiStep(baseLr, milestones, gamma)` implemented
- [ ] `LrScheduler.exponential(baseLr, gamma)` implemented
- [ ] `LrScheduler.withWarmup(base, baseLr, warmupSteps)` composable wrapper
- [ ] All schedulers are stateless (pure function of step)
- [ ] Existing cosineWarmup and constant unchanged
- [ ] Unit test: StepLR decays at correct intervals
- [ ] Unit test: MultiStepLR decays at each milestone
- [ ] Unit test: warmup wrapper produces linear ramp then delegates

## References

- [ResNet Paper (He et al., 2015)](https://arxiv.org/abs/1512.03385) — Uses StepLR with step=30, gamma=0.1
- [PyTorch LR Schedulers](https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate)

**Comment by Samyssmile** [comment]
Implemented in 0d83e53: StepDecayScheduler, MultiStepScheduler, ExponentialDecayScheduler, and WarmupScheduler — all as stateless Java records with factory methods on LrScheduler.
